app/lib/server/lib/sendNotificationsOnMessage.js

```diff
@@ -77,6 +78,10 @@ export const sendNotification = async ({
 
 	const isHighlighted = messageContainsHighlight(message, subscription.userHighlights);
 
+	if (hasMentionToUser || isHighlighted || (!disableAllMessageNotifications && (hasMentionToAll || hasMentionToHere))) {
```

Comment:
`hasMentionToUser` is also true for thread replies, which should not generate a mention record.